supervisor/supervisor.go

```diff
@@ -143,8 +143,8 @@ func (s *Supervisor) httpRequest(body string) error {
 
 	res.Body.Close()
 
-	if res.StatusCode != 200 {
-		return fmt.Errorf("Non-200 status code received")
+	if res.StatusCode < http.StatusOK || res.StatusCode > http.StatusIMUsed {
```

Comment:
I did not want to use 299 as the maximum value since the codes after 226 are unassigned and maybe is not too idiomatic to allow nonexistent values.